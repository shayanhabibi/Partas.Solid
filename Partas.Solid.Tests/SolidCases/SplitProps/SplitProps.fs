module Partas.Solid.Tests.SolidCases.SplitProps

open Partas.Solid
open Fable.Core

type [<Erase>] SplitProps() =
    inherit RegularNode()
    [<SolidTypeComponent>]
    member props.SplitProps =
        div(class' = props.class') {
            button(draggable = props.draggable)
        }

type [<Erase>] NestedSplitProps() =
    inherit RegularNode()
    [<SolidTypeComponent>]
    member props.SplitProps =
        div() {
            button(class' = props.class')
            div() {
                button(onClick = (fun _ -> System.Console.WriteLine props.class'))
            }
            "text"
        }

    