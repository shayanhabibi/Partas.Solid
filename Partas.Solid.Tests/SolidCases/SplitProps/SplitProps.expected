import { splitProps } from "solid-js";

export function SplitProps(props) {
    const [PARTAS_LOCAL, PARTAS_OTHERS] = splitProps(props, ["class", "draggable"]);
    return <div class={PARTAS_LOCAL.class}>
        <button draggable={PARTAS_LOCAL.draggable} />
    </div>;
}

export function NestedSplitProps(props) {
    const [PARTAS_LOCAL, PARTAS_OTHERS] = splitProps(props, ["class"]);
    return <div>
        <button class={PARTAS_LOCAL.class} />
        <div>
            <button onClick={(_arg) => {
                    console.log(PARTAS_LOCAL.class);
                }} />
        </div>
        text
    </div>;
}

